### PR TITLE
Added PrivateResources as config object

### PR DESCRIPTION
**Pull request by elburgl69**
Make the ACL more flexible bij having the private resources as a seperate config.

### CODE REVIEW

File: app/config/services.php [hunk 1]
```diff
@@ -25,6 +25,11 @@
         $config->merge($override);
     }
     
+    if (is_readable(APP_PATH . '/config/privateResources.php')) {
```
Comment (sergeyklay) [inline app/config/services.php]: How about put this in Di as separated object?

File: app/library/Acl/Acl.php [hunk 2]
```diff
@@ -32,26 +32,7 @@ class Acl extends Component
...
-            'search',
-            'edit',
-            'create',
-            'delete'
-        ],
-        'permissions' => [
-            'index'
-        ]
-    ];
+    private $privateResources = array();
```
Comment (sergeyklay) [inline app/library/Acl/Acl.php]:
```php
private $privateResources = [];
```
We are in the 21st century

File: app/library/Acl/Acl.php [hunk 3]
```diff
@@ -246,4 +227,17 @@ protected function getFilePath()
         return $this->filePath;
     }
 
+    /**
+     * Adds an array of private resources to the ACL object.
+     * 
+     * @param array $resources
+     */
+    public function addPrivateResources($resources) {
```
Comment (sergeyklay) [inline app/library/Acl/Acl.php]:
```php
public function addPrivateResources(array $resources)
{
    // ...
}

File: app/library/Acl/Acl.php (hunk 2 above)
Comment (elburgl69) [inline app/library/Acl/Acl.php]: Done.

File: app/config/services.php (hunk 1 above)
Comment (elburgl69) [inline app/config/services.php]: Done

File: app/library/Acl/Acl.php (hunk 3 above)
Comment (elburgl69) [inline app/library/Acl/Acl.php]: Done

File: app/config/services.php [hunk 4]
```diff
@@ -158,13 +153,23 @@
 });
 
 /**
+ * Setup the private resources, if any, for performance optimization of the ACL.  
+ */
+$di->setShared('PrivateResources', function() {
```
Comment (sergeyklay) [inline app/config/services.php]: How about `AclResources `

File: app/config/services.php [hunk 5]
```diff
@@ -158,13 +153,23 @@
...
+
+/**
  * Access Control List
  * Reads privateResource as an array from the config object.
  */
 $di->set('acl', function () {
     $acl = new Acl();
-    $config = $this->getConfig();
-    $pr=$config->privateResources->toArray();
+    $pr = $this->getShared('PrivateResources')->privateResources->toArray();
```
Comment (sergeyklay) [inline app/config/services.php]: How about:
```php
$pr = $this->getShared('AclResources')->privateResources->toArray();
```

File: app/library/Acl/Acl.php [hunk 6]
```diff
@@ -232,7 +232,7 @@ protected function getFilePath()
      * 
      * @param array $resources
      */
-    public function addPrivateResources($resources) {
+    public function addPrivateResources(array $resources) {
         if (count($resources) > 0) {
```
Comment (sergeyklay) [inline app/library/Acl/Acl.php]: Now we can safely remove the condition
```php
if (count($resources) > 0) {
```

File: app/config/services.php [hunk 7]
```diff
@@ -153,10 +153,25 @@
...
+});
+
+/**
  * Access Control List
+ * Reads privateResource as an array from the config object.
  */
 $di->set('acl', function () {
-    return new Acl();
+    $acl = new Acl();
+    $pr = $this->getShared('PrivateResources')->privateResources->toArray();
```
Comment (sergeyklay) [inline app/config/services.php]:
```php
$pr = $this->getShared('AclResources')->privateResources->toArray();
```

**Comment by sergeyklay** [comment]
@lburgl69 Thank you for contributing. There are a few little things. Could you please take a look

**Comment by sergeyklay** [comment]
Thanks

**Pull request merged**